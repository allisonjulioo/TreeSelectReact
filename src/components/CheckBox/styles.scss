.checkbox-input {
  height: 40px;
  cursor: pointer;
  display: flex;
  align-items: center;
  user-select: none;
  &:hover {
    background: cornsilk;
  }
}
