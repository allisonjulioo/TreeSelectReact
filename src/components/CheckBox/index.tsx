import React, { PureComponent } from 'react';
import './styles.scss';

export default class extends PureComponent {
  render() {
    return <input type="checkbox" id="checkbox" className="checkbox-input" />;
  }
}
