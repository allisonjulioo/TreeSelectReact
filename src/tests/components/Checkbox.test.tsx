/* eslint-disable no-undef */
import React from "react";
import CheckBox from "../../components/CheckBox/index";
import { shallow } from "enzyme";

const renderComponent = (props = {}) => shallow(<CheckBox />);

describe("<CheckBox/>", () => {
  beforeEach(() => {});
  it("renders the component", () => {
    const component = shallow(<CheckBox />);
    expect(component).toMatchSnapshot();
  });
  it("should match the snapshot", () => {
    expect(CheckBox).toMatchSnapshot();
  });
  it("should checkbox alredy changed to checked", () => {
    const wrapper = renderComponent();
    wrapper.simulate("change", { target: { checked: false } });
  });
});
