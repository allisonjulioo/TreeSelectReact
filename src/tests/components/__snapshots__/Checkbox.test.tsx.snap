// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox/> renders the component 1`] = `
<input
  className="checkbox-input"
  type="checkbox"
/>
`;

exports[`<CheckBox/> should match the snapshot 1`] = `[Function]`;
